lib/Fmt_ast.ml

```diff
@@ -1366,7 +1409,7 @@ and fmt_label_arg ?(box = true) ?epi ?parens ?eol c
   | _ -> fmt_label lbl ":@," $ fmt_expression c ~box ?epi ?parens xarg
 
 and expression_width c xe =
-  String.length (Cmts.preserve (fun () -> fmt_expression c xe) c.cmts)
+  String.length (Cmts.preserve c (fun () -> fmt_expression c xe))
```

Comment:
Out of curiosity, even if the issue was not reproduced on expressions, wouldn't it be better to cache the results as well for consistency, and just in case it breaks in the future?

Reply:
That takes more memory for no known reason. I thought about adding a warning when we detect > 2 nested calls of `expression_width` (which would indicate there's a way to nest arbitrarily - I haven't found a way to do that) but that's not super useful. I just pushed a commit that does that.